### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by sameedmohsin28**
In this PR, I have

Added the following validations:
- For the User model:
  - Name must not be blank.
  - PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
  - Title must not be blank.
  - Title must not exceed 250 characters.
  - CommentsCounter must be an integer greater than or equal to zero.
  - LikesCounter must be an integer greater than or equal to zero.

Added unit specs for all of the models' methods and validations.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      @comment = Comment.new(text: 'comment1', user_id: @user.id, post_id: @post.id)
+    end
+
+    it 'Should update comments counter' do
+      @comment.save
+      @post.reload
+      expect(@post.comments_counter).to eq(1)
+    end
+  end
+end
```
Comment (Shedrack-Sunday) [inline spec/models/comment_spec.rb]: - [ ] Good job 👏 . However,  all your Model specs are meant to  have  Validation tests as specified in the project requirements.

### Sample code:
```
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```
### Instruction:
image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @sameedmohsin28        👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with rails structured format✔️    
 
  
# Required Changes  🌵  

- [ ] You've done well, However, kindly check some key point note I pointed out to meet the requirements before moving to the next project 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (sameedmohsin28) [reply spec/models/comment_spec.rb]: Could you kindly elaborate further on what you mean by that? I can not understand that.

Comment (Shedrack-Sunday) [reply spec/models/comment_spec.rb]: Hello @sameedmohsin28 , i clearly gave a sample code to show you what's needed. You need to add testing for all validation made to the models. 
- The one I added  showed you a spec testing for the name validation which you added in the user model.

Comment (sameedmohsin28) [reply spec/models/comment_spec.rb]: I have already added the tests for all the validations in my opinion which includes the example you gave. That's why I asked if I was missing out on something. Below is the screenshot which contains name validation for User in highlighted text. Please tell me what else do I need to add more.

image

Comment (Shedrack-Sunday) [reply spec/models/comment_spec.rb]: Please use the `describe` rspec grouping for the validations from the requirements. Check the sample I gave. Add same for all other files, likes, post and users.

Comment (sameedmohsin28) [reply spec/models/comment_spec.rb]: Oh ok. I get it. Thanks!

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @sameedmohsin28  

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

image


Highlights
- [x] Good code organization ✔️
- [x] Nice use of ruby features ✔️ 
- [x] Kudus on meeting the requirements as suggested previously  ✔️ 

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**